File: app/src/main/java/com/grace/placessearch/common/ui/injection/component/ActivityComponent.java
package com.grace.placessearch.common.ui.injection.component;

import com.grace.placessearch.common.app.injection.component.PlacesSearchComponent;
import com.grace.placessearch.maps.ui.VenuesMapActivity;
import com.grace.placessearch.search.ui.SearchActivity;
import com.grace.placessearch.common.ui.injection.scope.ActivityScope;
import com.grace.placessearch.venue.detail.ui.VenueDetailsActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = PlacesSearchComponent.class)
public interface ActivityComponent {

    void inject(SearchActivity activity);

    void inject(VenuesMapActivity activity);

    void inject(VenueDetailsActivity activity);
}

File: app/src/main/kotlin/com/grace/placessearch/common/ui/injection/component/ActivityComponent.kt
package com.grace.placessearch.common.ui.injection.component

import com.grace.placessearch.common.app.injection.component.PlacesSearchComponent
import com.grace.placessearch.common.ui.injection.scope.ActivityScope
import com.grace.placessearch.maps.ui.VenuesMapActivity
import com.grace.placessearch.search.ui.SearchActivity
import dagger.Component

@ActivityScope
@Component(dependencies = [PlacesSearchComponent::class])
interface ActivityComponent {

    fun inject(activity: SearchActivity)

    fun inject(activity: VenuesMapActivity)

}
